src/main.cpp

```diff
@@ -314,6 +316,27 @@ int eoserv_main(int argc, char *argv[])
 			}
 		}
 
+		if (static_cast<int>(config["ThreadPoolThreads"]) <= 0)
+		{
+			if (std::thread::hardware_concurrency() == 0)
+			{
+				config["ThreadPoolThreads"] = static_cast<int>(util::ThreadPool::DEFAULT_THREADS);
+			}
+			else
+			{
+				config["ThreadPoolThreads"] = static_cast<int>(std::thread::hardware_concurrency());
+			}
+		}
+		else if (static_cast<size_t>(config["ThreadPoolThreads"].GetInt()) > util::ThreadPool::MAX_THREADS)
+		{
+			Console::Wrn("Value of ThreadPoolThreads is too high. Overriding user-defined threadpool threads with %d", util::ThreadPool::MAX_THREADS);
+			config["ThreadPoolThreads"] = static_cast<int>(util::ThreadPool::MAX_THREADS);
+		}
+
+		size_t threadPoolThreads = static_cast<size_t>(static_cast<int>(config["ThreadPoolThreads"]));
```

Comment:
line 330: `static_cast<size_t>(config["ThreadPoolThreads"].GetInt())`
line 366: `static_cast<size_t>(static_cast<int>(config["ThreadPoolThreads"]))`

Could we just do `static_cast<size_t>(config["ThreadPoolThreads"])`?

Reply:
What the heck was I writing...haha I'll make it more consistent.

Reply:
I remember now. This doesn't work because there is no implicit operator conversion from `util::variant` to `size_t`. So this is actually intentional - it needs to be manually converted to `int` first and then to `size_t`. I can still make it more consistent though.